### PR TITLE
[ISSUE #6386] Some improvements for compactionTopic

### LINKED ISSUE #6386: Some improvements for compactionTopic
The issue tracker is used for bug reporting purposes **ONLY** whereas feature request needs to follow the [RIP process](https://github.com/apache/rocketmq/wiki/RocketMQ-Improvement-Proposal). To avoid unnecessary duplication, please check whether there is a previous issue before filing a new one.

It is recommended to start a discussion thread in the [mailing lists](http://rocketmq.apache.org/about/contact/) or [github discussions](https://github.com/apache/rocketmq/discussions) in cases of discussing your deployment plan, API clarification, and other non-bug-reporting issues.
We welcome any friendly suggestions, bug fixes, collaboration, and other improvements.

Please ensure that your bug report is clear and self-contained. Otherwise, it would take additional rounds of communication, thus more time, to understand the problem itself.

Generally, fixing an issue goes through the following steps:
1. Understand the issue reported;
1. Reproduce the unexpected behavior locally;
1. Perform root cause analysis to identify the underlying problem;
1. Create test cases to cover the identified problem;
1. Work out a solution to rectify the behavior and make the newly created test cases pass;
1. Make a pull request and go through peer review;

As a result, it would be very helpful yet challenging if you could provide an isolated project reproducing your reported issue. Anyway, please ensure your issue report is informative enough for the community to pick up. At a minimum, include the following hints:

**FEATURE REQUEST**

1. Please describe the feature you are requesting.

 -- Graceful shutdown needs to wait for the compactionTopic to build the index
 -- Exception handling that may occur when adding pullMessageFromMaster
 -- Update usage docs

2. Provide any additional detail on your proposed use case for this feature.

3. Indicate the importance of this issue to you (blocker, must-have, should-have, nice-to-have). Are you currently using any workarounds to address this issue?

4. If there are some sub-tasks involved, use -[] for each sub-task and create a corresponding issue to map to the sub-task:

- [sub-task1-issue-number](example_sub_issue1_link_here): sub-task1 description here, 
- [sub-task2-issue-number](example_sub_issue2_link_here): sub-task2 description here,
- ...

### PR DESCRIPTION
**Pull request by guyinyou**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

fix #6386 

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by RongtongJin** [comment]
Error:  /home/runner/work/rocketmq/rocketmq/store/src/main/java/org/apache/rocketmq/store/kv/CompactionService.java:129:42: '{' is not preceded with whitespace. [WhitespaceAround]
Error:  /home/runner/work/rocketmq/rocketmq/store/src/main/java/org/apache/rocketmq/store/kv/MessageFetcher.java:149:13: 'if' is not followed by whitespace. [WhitespaceAround]
Error:  /home/runner/work/rocketmq/rocketmq/store/src/main/java/org/apache/rocketmq/store/kv/MessageFetcher.java:149:67: '{' is not preceded with whitespace. [WhitespaceAround]
Audit done.

**Comment by guyinyou** [comment]
> Error: /home/runner/work/rocketmq/rocketmq/store/src/main/java/org/apache/rocketmq/store/kv/CompactionService.java:129:42: '{' is not preceded with whitespace. [WhitespaceAround] Error: /home/runner/work/rocketmq/rocketmq/store/src/main/java/org/apache/rocketmq/store/kv/MessageFetcher.java:149:13: 'if' is not followed by whitespace. [WhitespaceAround] Error: /home/runner/work/rocketmq/rocketmq/store/src/main/java/org/apache/rocketmq/store/kv/MessageFetcher.java:149:67: '{' is not preceded with whitespace. [WhitespaceAround] Audit done.

fix done

**Review by fuyou001 — DISMISSED** [review]
LGTM

**Review by RongtongJin — APPROVED** [review]

**Review by panzhi33 — APPROVED** [review]

File: docs/cn/Example_Compaction_Topic_cn.md [hunk 1]
```diff
@@ -1,59 +1,73 @@
 # Compaction Topic
 
 ## 使用方式
+
+### 打开namesrv上支持顺序消息的开关
+
```
Comment (ltamber) [inline docs/cn/Example_Compaction_Topic_cn.md:6]: maybe we need to explain why order message is necessary here.

File: docs/cn/Example_Compaction_Topic_cn.md
Comment (ltamber) [inline docs/cn/Example_Compaction_Topic_cn.md:1]: the [en edition](https://github.com/apache/rocketmq/blob/develop/docs/en/Example_Compaction_Topic.md) example file should update in sync.

File: docs/cn/Example_Compaction_Topic_cn.md [hunk 2]
```diff
@@ -1,59 +1,73 @@
...
+        Message msg=new Message(topic,tag,key,"bodys"getBytes(StandardCharsets.UTF_8));
+        SendResult sendResult=producer.send(msg,(mqs,message,shardingKey)->{
+        int select=Math.abs(shardingKey.hashCode());
+        if(select< 0){
+        select=0;
+        }
+        return mqs.get(select%mqs.size());
+        },key);
+
+        System.out.printf("%s%n",sendResult);
```
Comment (ltamber) [inline docs/cn/Example_Compaction_Topic_cn.md:40]: the code format looks messed up

File: docs/cn/Example_Compaction_Topic_cn.md [hunk 3]
```diff
@@ -1,59 +1,73 @@
...
+        }
+        });
+
+        Map<String, byte[]>kvStore=Maps.newHashMap();
+        while(true){
+        List<MessageExt> msgList=consumer.poll(1000);
+        if(msgList!=null){
+        msgList.forEach(msg->kvStore.put(msg.getKeys(),msg.getBody()));
+        }
+        }
```
Comment (ltamber) [inline docs/cn/Example_Compaction_Topic_cn.md:70]: the code format looks messed up

Comment (guyinyou) [reply docs/cn/Example_Compaction_Topic_cn.md]: fixed

File: docs/cn/Example_Compaction_Topic_cn.md
Comment (guyinyou) [reply docs/cn/Example_Compaction_Topic_cn.md:1]: added

**Review by fuyou001 — APPROVED** [review]

**Pull request merged**